Return HandlerResult instead of Publisher<HandlerResult> in HttpHandlerAdapter#handle()

sdeleuze opened the pull request:
As discussed during the meeting, this PR changes `HttpHandlerAdapter#handle()` from returning `Publisher<HandlerResult>` to `HandlerResult`.

This change implies another one from my POV: for non `@ResponseBody` based `HttpHandler`, the handling has been moved from `HttpHandlerAdapter` to an `HandlerResultHandler` implementation (the new `HttpHandlerResultHandler` class). That seems to me a logical change: performing the request is a `HandlerResultHandler`responsibility, not a `HttpHandlerAdapter` responsibility, for all implementations.

rstoyanchev commented on src/main/java/org/springframework/reactive/web/dispatch/handler/HttpHandlerAdapter.java:
```diff
@@ -44,10 +42,8 @@ public boolean supports(Object handler) {
-		Publisher<Void> publisher = httpHandler.handle(request, response);
-		return Streams.wrap(publisher).map(aVoid -> null);
+	public HandlerResult handle(ServerHttpRequest request, ServerHttpResponse response, Object handler) {
+		return new HandlerResult(handler, null);
```
This doesn't invoke the handler, which is the HandlerAdapter's main purpose. This should be:
````
Publisher<Void> completion = httpHandler.handle(request, response);
return new HandlerResult(handler, completion);
````

Then add a simple HandlerResultHandler for `Publisher<Void>`

rstoyanchev commented on src/main/java/org/springframework/reactive/web/dispatch/HttpHandlerResultHandler.java:
```diff
@@ -0,0 +1,47 @@
+/**
+ * @author Sebastien Deleuze
+ */
+public class HttpHandlerResultHandler implements Ordered, HandlerResultHandler {
```
HandlerResultHandler should be handling the result of invoking a handler, not invoking the handler. All you need here is a HandlerResultHandler that supports `Publisher<Void>`.

sdeleuze commented:
@rstoyanchev Thanks for your feedback, I have taken in account your remarks in the updated commit.

rstoyanchev commented:
Looks good, only minor comment that `HttpHandlerResultHandler` should probably be something like `SimpleHandlerResultHandler`. Consider that Publisher<Void> generally as a handler return value regarding of what handler was executed (e.g. HttpHandler, annotated controller method).

sdeleuze commented:
Ok, I have renamed it to `SimpleHandlerResultHandler`, updated the `SimpleHandlerResultHandler#supports()` method to not restrict it to `HttpHandler` and updated `ResponseBodyResultHandler#supports()` to not support handler methods returning `Publisher<Void>` since this case will be handle by `SimpleHandlerResultHandler` as described in your comment.

sdeleuze commented:
Merged
